### PR TITLE
Fix a whole bunch of frontend test issues

### PR DESCRIPTION
**Pull request by nickclyde**
## Related Issue or Background Info

- We are often having to rerun our frontend tests in CI 3-4 times to get them to pass.
- The test output is also very verbose, lots of unuseful and noisy warnings come out

## Changes Proposed

(Here is [a link to view changes](https://github.com/CDCgov/prime-simplereport/pull/3028/files?authenticity_token=GY68fXJ9pAuDpqcTU1zQU5iiN1rLF1DLHTWLOJq8tKMLdWVvAp%2Bq9%2BtofGWPkdYnBMLj9jSDC1Ngkr3Jj4Mlag%3D%3D&file-filters%5B%5D=.gradle&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.tsx&hide-deleted-files=true&w=1#diff-bc6abb0836b5c207b4232e7db185278d9323082d639a06742779d251935692f4) with whitespace and deleted files filtered out)

- Added [eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library) to our eslint config (ESLint plugin to follow best practices and anticipate common mistakes when writing tests with Testing Library)
  - Fix the ~400ish errors and warnings from that plugin. Most of these have to do with the way we've been using `waitFor` and `act` in our tests. The plugin suggests to never use `act` in your tests because Testing Library's methods actually have `act` built in, and the warning appearing in your tests actually means that there was a side effect in your test that was not explicitly anticipated by your expectations. A lot of times this can be resolved by using `waitForElementToBeRemoved` to explicitly wait for a loading screen to pass, or for a button to be re-enabled, for example. More info here: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily
- Added [eslint-plugin-unused-imports](https://github.com/sweepline/eslint-plugin-unused-imports) to enable automatically removing unused imports when `eslint --fix` is ran, meaning no more lint commits of shame!
  - Also removed some unused vars from throughout the codebase
- Added [eslint-plugin-jest-dom](https://github.com/testing-library/eslint-plugin-jest-dom) which catches a few more common hangups in tests that use jest-dom
- Remove unnecessary calls to `cleanup` in tests
- Converted all tests still using `react-test-renderer` to use `testing-library` instead
- Converted some snapshot tests into tests with actual expectations (some snapshot tests still remain)
- Managed to get rid of all console errors and warnings during test runs
  - Suppressing the deprecation warnings from moment
  - Suppress a random `DOMException` from `QueueItem.test.tsx`
- Bump the global Jest timeout to 30s. This ended up being what was causing our CI runs to fail, as explained by [this StackOverflow comment](https://stackoverflow.com/questions/60504720/jest-cannot-read-property-createevent-of-null#comment121546358_66008721)

### UI
- [ ] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: frontend/src/app/Settings/Users/UsersSideNav.tsx [hunk 1]
```diff
@@ -41,6 +42,12 @@ const UsersSideNav: React.FC<Props> = ({
                   "padding-105 padding-right-2",
                   activeUserId === user.id && "usa-current"
                 )}
+                aria-selected={activeUserId === user.id}
+                aria-label={displayFullName(
```
Comment (nickclyde) [inline frontend/src/app/Settings/Users/UsersSideNav.tsx:46]: Just added these for use in tests

File: frontend/src/app/testQueue/AoEForm/AoEForm.tsx [hunk 2]
```diff
@@ -208,7 +208,7 @@ const AoEForm: React.FC<Props> = ({
             </p>
             {emails.map((email) => (
               <span
-                key={"test-result-delivery-preference-email"}
+                key={`test-result-delivery-preference-email-${email}`}
```
Comment (nickclyde) [inline frontend/src/app/testQueue/AoEForm/AoEForm.tsx:211]: this ensures unique `key` props to suppress console error

File: frontend/package.json [hunk 3]
```diff
@@ -124,7 +126,18 @@
...
       "import/newline-after-import": 1,
-      "import/no-commonjs": 0
+      "import/no-commonjs": 0,
+      "no-unused-vars": "off",
+      "unused-imports/no-unused-imports": "error",
+      "unused-imports/no-unused-vars": [
+        "warn",
+        {
+          "vars": "all",
+          "varsIgnorePattern": "^_",
```
Comment (nathancrtr) [inline frontend/package.json:136]: Ooh, does this mean I can prefix unused variables with an underscore and the linter won't yell at me for it? 😁

Comment (nickclyde) [reply frontend/package.json:136]: Yes!! :tada:

**Review by nathancrtr — APPROVED** [review]
The patterns introduced here result in faster, less-fragile runs with no console output spam. This is a big win! 🎉  🚀

**Review by emmastephenson — APPROVED** [review]
LGTM, thank you so much for all your hard work on this one Nick!!

**Review by zdeveloper — APPROVED** [review]
LGTM! thank you for the hard work that went into this

**Pull request merged**